weather-app/src/components/Searchbar.vue

```diff
@@ -1,33 +1,261 @@
+<script>
+import axios from 'axios'
+let reAlpha = /^[a-z]+$/i
+
+export default {
+    
+    name: 'Searchbar',
+    data() {
+        return {
+            countries: null,
+            cities: null,
+            cityShow: true,
+            countryShow: true,
+            weatherShow: false,
+            fahrenheit: true,
+            // dom variables
+            cityName: null,
+            currentTemp: null,
+            maxTemp: null,
+            minTemp: null,
+            tempSymbol: null,
+            weatherImageUrl: null,
+            errorMessage: null,
+            country: null,
+            state: null
```

Comment:
If you have control over the data like you do here, in general, prefer empty / uninitialized versions of the data types your variables will have over `null`. For instance:

```js
country: "",
state: "",
minTemp: 0,
maxTemp: 0,
country: {},
countries: []
```

This implicitly declares the data type of the variable you are expecting each of the values to hold. In addition, you can get away with fewer `null` checks in your codebase which improves clarity, readability, and reduces errors (You are now no longer potentially operating on null values)